lib/Serialization/Deserialization.cpp

```diff
@@ -4555,6 +4504,93 @@ void ModuleFile::loadAllMembers(Decl *container, uint64_t contextData) {
   }
 }
 
+static Decl* handleErrorAndFudgeMissingMember     (llvm::Error &&,  ASTContext &,  Decl*         container);
+static Decl* handleErrorAndFudgeMissingClassMember(llvm::Error &&,  ASTContext &,  ClassDecl*    container);
+static Decl* handleErrorAndFudgeMissingProtoMember(llvm::Error &&,  ASTContext &,  ProtocolDecl* container);
+static Decl* handleErrorAndFudgeMissingMiscMember (llvm::Error &&);
+
+Decl* ModuleFile::getOrFudgeOneMember(Decl *container, DeclID rawID) {
+  Expected<Decl *> next = getDeclChecked(rawID);
+  
+  if (next) {
+    assert(next.get() && "unchecked error deserializing next member");
+    return next.get();
+  }
+  llvm::Error &&error = next.takeError();
+  if (!getContext().LangOpts.EnableDeserializationRecovery)
+    fatal(std::move(error));
+
+  return handleErrorAndFudgeMissingMember(std::move(error), getContext(), container);
+}
+
+
+Decl* handleErrorAndFudgeMissingMember(llvm::Error &&error, ASTContext& context, Decl *container) {
+  auto *containingClass = dyn_cast<   ClassDecl>(container);
```

Comment:
The pattern in the existing code is the more-standard LLVM style for dynamic type-switching:

if (auto *foo = dyn_cast<...>) { ... }
else if (auto *bar = dyn_cast<...>) { ... }